src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/CombineAttributeArrays.cpp

```diff
@@ -0,0 +1,210 @@
+            arrayOffset += inputArrays[j - 1]->getNumberOfComponents();
+          }
+          for(size_t k = 0; k < numComps; k++)
+          {
+            if(maxValues[arrayOffset + k] == minValues[arrayOffset + k])
+            {
+              outputArray[stackedDims * i + (arrayOffset) + k] = static_cast<DataType>(0);
+            }
+            else
+            {
+              outputArray[stackedDims * i + (arrayOffset) + k] = (inputArray[numComps * i + k] - minValues[arrayOffset + k]) / (maxValues[arrayOffset + k] - minValues[arrayOffset + k]);
+            }
+          }
+        }
+        arrayOffset = 0;
+      }
+    }
+    else
+    {
+      size_t outputNumComps = outputArray.getNumberOfComponents();
+      size_t compsWritten = 0;
+      for(const auto* inputArrayPtr : inputArrays)
+      {
+        const InputArrayType& inputArray = *inputArrayPtr; // Get a reference var to the current input array
+        size_t numInputComps = inputArray.getNumberOfComponents();
+
+        for(size_t tupleIndex = 0; tupleIndex < numTuples; tupleIndex++)
+        {
+          for(size_t compIndex = 0; compIndex < numInputComps; compIndex++)
+          {
+            outputArray[tupleIndex * outputNumComps + compsWritten + compIndex] = inputArray[tupleIndex * numInputComps + compIndex];
+          }
+        }
+        compsWritten += numInputComps;
+      }
+    }
+    return {};
+  }
+
+public:
+  CombineAttributeArraysImpl(const CombineAttributeArraysImpl&) = delete; // Copy Constructor Not Implemented
+  void operator=(const CombineAttributeArraysImpl&) = delete;             // Move assignment Not Implemented
+};
+
+// -----------------------------------------------------------------------------
+CombineAttributeArrays::CombineAttributeArrays(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                                               CombineAttributeArraysInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+CombineAttributeArrays::~CombineAttributeArrays() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& CombineAttributeArrays::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> CombineAttributeArrays::operator()()
+{
+  if(m_ShouldCancel)
+  {
+    return {};
+  }
+  std::vector<DataObject*> inputArrays;
+  for(const auto& dataPath : m_InputValues->SelectedDataArrayPaths)
+  {
+    inputArrays.push_back(m_DataStructure.getData(dataPath));
+  }
+
+  auto& outputArray = m_DataStructure.getDataRefAs<IDataArray>(m_InputValues->StackedDataArrayPath);
+  auto dataType = outputArray.getDataType();
+  switch(dataType)
```

Comment:
@imikejackson So this is fine to leave as is if you want but I recently discovered the `ExecuteDataFunction` in `FilterUtilities.hpp` which I believe would let you condense this whole switch statement into one line with a few tweaks to the Impl class you are using to do the actual work.  The CropVertexGeometryFilter has an example of it being used.